node/src/service.rs

```diff
@@ -358,6 +371,13 @@ where
 			.spawn("trace-filter-cache", trace_filter_task);
 	}
 
+	// Spawn debug task if enabled.
+	if let Some(debug_task) = debug_task {
+		task_manager
+			.spawn_essential_handle()
+			.spawn("ethapi-debug", debug_task);
```

Comment:
If I understand correctly, you want to spawn this in a separate thread.
For this to work, you should use `spawn_blocking`

Reply:
This task is responsible to read requests from the channel, then itself call `spawn_blocking` and await for the result to send it in a response channel.